File: srcs/backend/src/schemas/common/responses.js
/**
 * @brief Common response schemas for ft_transcendence backend
 * 
 * @description Standard response formats used across all routes
 */

export default [
  // Success response schema
  {
    $id: 'SuccessResponse',
    type: 'object',
    properties: {
      success: { 
        type: 'boolean', 
        description: 'Operation success status',
        example: true
      },
      message: { 
        type: 'string', 
        description: 'Human-readable response message',
        example: 'Operation completed successfully'
      }
    },
    required: ['success', 'message']
  },

  // Error response schema
  {
    $id: 'ErrorResponse',
    type: 'object',
    properties: {
      success: { 
        type: 'boolean', 
        description: 'Operation success status (always false)',
        example: false
      },
      message: { 
        type: 'string', 
        description: 'Error description',
        example: 'Operation failed'
      },
      error: { 
        type: 'object',
        description: 'Detailed error information',
        properties: {
          code: { 
            type: 'string',
            example: 'VALIDATION_ERROR'
          },
          details: { 
            type: 'string',
            example: 'Invalid input provided'
          }
        }
      }
    },
    required: ['success', 'message']
  },

  // Validation error response schema
  {
    $id: 'ValidationError',
    type: 'object',
    properties: {
      success: { 
        type: 'boolean', 
        example: false
      },
      message: { 
        type: 'string',
        example: 'Validation failed'
      },
      error: {
        type: 'object',
        properties: {
          code: { 
            type: 'string',
            example: 'VALIDATION_ERROR'
          },
          details: {
            type: 'array',
            items: {
              type: 'object',
              properties: {
                field: { type: 'string' },
                message: { type: 'string' }
              }
            }
          }
        }
      }
    },
    required: ['success', 'message']
  }
]

File: srcs/ai/src/utils/responses.py
from datetime import datetime, UTC
from typing import Any, Optional

def success_response(data: Any) -> dict:
    """Create standardized success response.

    Args:
        data: Response data payload

    Returns:
        Standardized success response dict
    """
    return {
        "success": True,
        "data": data,
        "error": None,
        "timestamp": datetime.now(UTC).isoformat()
    }

def error_response(code: str, message: str, details: Optional[dict] = None) -> dict:
    """Create standardized error response.

    Args:
        code: Error code (e.g., "INVALID_IMAGE")
        message: Human-readable error message
        details: Optional additional error details

    Returns:
        Standardized error response dict
    """
    return {
        "success": False,
        "data": None,
        "error": {
            "code": code,
            "message": message,
            "details": details or {}
        },
        "timestamp": datetime.now(UTC).isoformat()
    }
